packages/block-editor/src/components/block-breadcrumb/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockTitle from '../block-title';
+
+/**
+ * Block breadcrumb component, displaying the hierarchy of the current block selection as a breadcrumb.
+ *
+ * @return {WPElement} Block Breadcrumb.
+ */
+const BlockBreadcrumb = function() {
+	const { selectBlock, clearSelectedBlock } = useDispatch( 'core/block-editor' );
+	const { clientId, parents, hasSelection } = useSelect( ( select ) => {
+		const {
+			getSelectionStart,
+			getSelectedBlockClientId,
+			getBlockParents,
+		} = select( 'core/block-editor' );
+		const selectedBlockClientId = getSelectedBlockClientId();
+		return {
+			parents: getBlockParents( selectedBlockClientId ),
+			clientId: selectedBlockClientId,
+			hasSelection: !! getSelectionStart().clientId,
+		};
+	}, [] );
+
+	/*
+	 * Disable reason: The `list` ARIA role is redundant but
+	 * Safari+VoiceOver won't announce the list otherwise.
+	 */
+	/* eslint-disable jsx-a11y/no-redundant-roles */
+	return (
+		<ul className="block-editor-block-breadcrumb" role="list" aria-label={ __( 'Block breadcrumb' ) }>
+			<li
+				className={ ! hasSelection ? 'block-editor-block-breadcrumb__current' : undefined }
+				aria-current={ ! hasSelection ? 'true' : undefined }
+			>
+				{ hasSelection && (
+					<Button
+						className="block-editor-block-breadcrumb__button"
+						isTertiary
+						onClick={ clearSelectedBlock }
+					>
+						{ __( 'Document' ) }
+					</Button>
+				) }
+				{ ! hasSelection && __( 'Document' ) }
+			</li>
+			{ parents.map( ( parentClientId ) => (
+				<li key={ parentClientId }>
+					<Button
+						className="block-editor-block-breadcrumb__button"
+						isTertiary
+						onClick={ () => selectBlock( parentClientId ) }
+					>
+						<BlockTitle clientId={ parentClientId } />
+					</Button>
+				</li>
+			) ) }
+			{ !! clientId && (
+				<li className="block-editor-block-breadcrumb__current" aria-current="true">
```

Comment:
What's the reason it's displayed as text? We keep the `Document` as a button all the time (as noted in https://github.com/WordPress/gutenberg/pull/17838#issuecomment-545840271) but here we have 2 different representations. Given that it moves focus back to the editor, it might be fine to keep the button so you could move the focus back when using the keyboard. 

There is also no visual difference between the clickable button and the text at the moment.

Reply:
I did a quick test to validate my idea with:
```diff
diff --git a/packages/block-editor/src/components/block-breadcrumb/index.js b/packages/block-editor/src/components/block-breadcrumb/index.js
index adc96ba4e..20e05c61d 100644
--- a/packages/block-editor/src/components/block-breadcrumb/index.js
+++ b/packages/block-editor/src/components/block-breadcrumb/index.js
@@ -66,7 +66,13 @@ const BlockBreadcrumb = function() {
                        ) ) }
                        { !! clientId && (
                                <li className="block-editor-block-breadcrumb__current" aria-current="true">
-                                       <BlockTitle clientId={ clientId } />
+                                       <Button
+                                               className="block-editor-block-breadcrumb__button"
+                                               isTertiary
+                                               onClick={ () => selectBlock( clientId ) }
+                                       >
+                                               <BlockTitle clientId={ clientId } />
+                                       </Button>
                                </li>
                        ) }
                </ul>
```

It doesn't work at all, because it doesn't move the focus to the content area the way I assumed. I think we should leave it as is and maybe tweak the visual to make it clear what is clickable and what is not.